src/PythonProjectBootstrapper/ProjectGenerationUtils.py

```diff
@@ -0,0 +1,204 @@
+# ----------------------------------------------------------------------
+# |
+# |  Copyright (c) 2024 Scientific Software Engineering Center at Georgia Tech
+# |  Distributed under the MIT License.
+# |
+# ----------------------------------------------------------------------
+import hashlib
+import itertools
+import os
+import sys
+from pathlib import Path
+
+from rich import print  # pylint: disable=redefined-builtin
+from rich.panel import Panel
+import yaml
+
+from dbrownell_Common import PathEx
+from PythonProjectBootstrapper import __version__
+
+# The following imports are used in cookiecutter hooks. Import them here to
+# ensure that they are frozen when creating binaries,
+import shutil  # pylint: disable=unused-import, wrong-import-order
+import textwrap  # pylint: disable=unused-import, wrong-import-order
+
+
+HASH_ALG = hashlib.sha256
+
+
+# ----------------------------------------------------------------------
+def _GenerateFileHash(filepath: Path, hash_fn) -> str:
+    PathEx.EnsureFile(filepath)
+    with open(filepath, "rb") as file:
+        hash_value = hashlib.file_digest(file, hash_fn).hexdigest()
+
+    return hash_value
+
+
+# ----------------------------------------------------------------------
+def _CreateManifest(generated_dir: Path) -> dict[str, str]:
+    manifest_dict: dict[str, str] = {}
+    generated_files: list[Path] = []
+
+    for root, _, files in os.walk(generated_dir):
+        generated_files += [Path(root) / Path(file) for file in files]
```

Comment:
Nit: As this code is written now, `Path(root)` is being computer for every iteration of `files`. Cacluate this value outside of the generator so that it is only converted once.